core/src/main/java/com/sequenceiq/cloudbreak/reactor/CollectDownscaleCandidatesHandler.java

```diff
@@ -40,17 +50,35 @@ public void accept(Event<CollectDownscaleCandidatesRequest> event) {
         CollectDownscaleCandidatesResult result;
         try {
             Stack stack = stackService.getByIdWithLists(request.getStackId());
-            Set<String> hostNames;
-            if (request.getHostNames() == null || request.getHostNames().isEmpty()) {
-                hostNames = ambariDecommissioner.collectDownscaleCandidates(stack, request.getHostGroupName(), request.getScalingAdjustment());
-            } else {
-                hostNames = request.getHostNames();
-                ambariDecommissioner.verifyNodeCount(stack, stack.getCluster(), hostNames.stream().findFirst().get());
-            }
-            result = new CollectDownscaleCandidatesResult(request, hostNames);
+            result = verifyRequestAndCreateResult(request, stack);
         } catch (Exception e) {
             result = new CollectDownscaleCandidatesResult(e.getMessage(), e, request);
         }
         eventBus.notify(result.selector(), new Event<>(event.getHeaders(), result));
     }
+
+    private CollectDownscaleCandidatesResult verifyRequestAndCreateResult(CollectDownscaleCandidatesRequest request, Stack stack) throws CloudbreakException {
+        CollectDownscaleCandidatesResult result;
+        if (isFqdnMissing(request) && Objects.isNull(request.getScalingAdjustment())) {
```

Comment:
Is it possible to catch this kind of error earlier? The best is somehow validate this during the REST request.